integration_tests/test_ibc.py

```diff
@@ -210,3 +211,89 @@ def test_ibc(cluster):
         assert [bal for bal in balances if int(bal["amount"]) > 0] == [
             {"amount": "10000000000", "denom": "basecro"},
         ]
+
+
+def test_update_chain_id(cluster):
+    data_root = next(iter(cluster.values())).data_root
+    # call chain-maind directly
+    raw = next(iter(cluster.values())).cosmos_cli().raw
+    relayer = ["relayer", "--home", data_root / "relayer"]
+
+    channels = json.loads(
+        raw(
+            "query",
+            "ibc",
+            "channel",
+            "channels",
+            node=cluster["ibc-1"].node_rpc(0),
+            output="json",
+        )
+    )["channels"]
+    assert channels[0]["state"] == "STATE_OPEN"
+    dst_channel = channels[0]["counterparty"]["channel_id"]
+    denom_hash = (
+        hashlib.sha256(f"transfer/{dst_channel}/basecro".encode()).hexdigest().upper()
+    )
+    # update chain-id
+    cluster["ibc-1"].stop_node()
+    genesis_data = json.loads(cluster["ibc-1"].export(0))
+    genesis_data["chain_id"] = "ibc-2"
+    upgrade_time = datetime.utcnow() + timedelta(seconds=5)
+    upgrade_time.replace(tzinfo=None).isoformat("T") + "Z"
+    genesis_data["genesis-time"] = str(upgrade_time)
+    cluster["ibc-1"].update_genesis(0, genesis_data)
+    cluster["ibc-1"].start_node(0)
+    rsp = json.loads(subprocess.check_output(relayer + ["chains", "list", "--json"]))
```

Comment:
I think relayer also need to update the chain-id in config accordingly.

Reply:
in my test no need to restart relayer, it works with the old chain-id, I'll test to update the new chain-id in the relayer config and restart the relayer to see the result.